handler/src/main/java/io/netty/handler/ipfilter/RuleBasedIpFilter.java

```diff
@@ -24,29 +24,41 @@
 import java.net.SocketAddress;
 
 /**
+ * <p>
  * This class allows one to filter new {@link Channel}s based on the
  * {@link IpFilterRule}s passed to its constructor. If no rules are provided, all connections
  * will be accepted.
+ * </p>
  *
+ * <p>
  * If you would like to explicitly take action on rejected {@link Channel}s, you should override
  * {@link AbstractRemoteAddressFilter#channelRejected(ChannelHandlerContext, SocketAddress)}.
+ * </p>
  */
 @Sharable
 public class RuleBasedIpFilter extends AbstractRemoteAddressFilter<InetSocketAddress> {
 
     private final IpFilterRule[] rules;
 
-    public RuleBasedIpFilter(IpFilterRule... rules) {
+    /**
+     * Create new Instance of {@link RuleBasedIpFilter} and filter incoming connections
+     * based on their IP address and {@code rules} applied.
+     *
+     * @param rules An array of {@link IpFilterRule} containing all rules.
+     * @throws NullPointerException If {@code rules} array or {@code rule} in
+     *                              {@code rules} array is {@code null}.
+     */
+    public RuleBasedIpFilter(IpFilterRule... rules) throws NullPointerException {
         this.rules = ObjectUtil.checkNotNull(rules, "rules");
+
+        for (IpFilterRule rule : this.rules) {
+            ObjectUtil.checkNotNull(rule, "rule");
```

Comment:
This is a non compatible change in behaviour imho...

Reply:
So how to tackle this? By removing `throws NullPointerException`?

Reply:
Since we're already throwing `NullPointerException`, I can remove `throws NullPointerException` and let it throw `NullPointerException` normally.

Reply:
It's sort of a strange behaviour that rule processing stops at the first null element.

You can scan the array to see if there are any nulls, and if so, replace the array with its first non-null sub-sequence. Assuming the mutability of the array is not part of the API. It's not clear.

Reply:
Yeah, I think this will work. We should remove `not null` rules.